source/vibe/db/postgresql/package.d

```diff
@@ -106,18 +107,19 @@ class __Conn : dpq2.Connection
     private void waitEndOfRead(in Duration timeout) // TODO: rename to waitEndOf + add FileDescriptorEvent.Trigger argument
     {
         import vibe.core.core;
+        import std.typecons: Unique;
 
-        auto sock = this.socket();
-
-        sock.blocking = false;
-        scope(exit)
+        version(Have_vibe_core)
         {
-            sock.blocking = true;
-            destroy(sock);
+            // vibe-core right now support only read trigger event
+            scope event = createFileDescriptorEvent(posixSocketDuplicate, FileDescriptorEvent.Trigger.read);
```

Comment:
`scope` keyword works strictly as
```D
scope(exit) destroy(event)
```
?
Can you rely on this?

What will work before for same variable: its `scope` declaration or `scope(exit)`?

Reply:
There should be auto, not scope my fault, createFileDescriptorEvent return struct not class, so yes there is no need for destroy here

Reply:
generaly is it better to not use scope(exit) it is implemented as try catch finally block which is not really good for performance

Reply:
Original code with comment:
```
scope(exit) destroy(event); // Prevents 100% CPU usage
```

Therefore also there was my question about `scope` - it looks more clear

Reply:
but scope in this case would not work AFAIK, from my test it seems that Unique works better than scope

Reply:
Ok `destroy` is also works ok